geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/set/AbstractSScanIntegrationTest.java

```diff
@@ -54,295 +74,397 @@ public void tearDown() {
-    jedis.sadd("a", "1", "12", "3");
-
+  public void givenSetWithThreeEntriesAndMatch_returnsOnlyMatchingElements() {
+    jedis.sadd(KEY, FIELD_ONE, FIELD_TWO, FIELD_THREE);
     ScanParams scanParams = new ScanParams();
     scanParams.match("1*");
 
-    ScanResult<byte[]> result =
-        jedis.sscan("a".getBytes(), "0".getBytes(), scanParams);
+    result = jedis.sscan(KEY.getBytes(), ZERO_CURSOR.getBytes(), scanParams);
 
     assertThat(result.isCompleteIteration()).isTrue();
-    assertThat(result.getResult()).containsExactlyInAnyOrder("1".getBytes(),
-        "12".getBytes());
+    assertThat(result.getResult()).containsOnly(FIELD_ONE.getBytes(),
+        FIELD_TWO.getBytes());
   }
 
   @Test
-  @SuppressWarnings("unchecked")
-  public void givenMultipleMatches_returnsMembersMatchingLastMatchParameter() {
-    jedis.sadd("a", "1", "12", "3");
+  public void givenSetWithThreeEntriesAndMultipleMatchArguments_returnsOnlyElementsMatchingLastMatchArgument() {
+    jedis.sadd(KEY, FIELD_ONE, FIELD_TWO, FIELD_THREE);
 
-    List<Object> result = (List<Object>) jedis.sendCommand("a", Protocol.Command.SSCAN, "a", "0",
-        "MATCH", "3*", "MATCH", "1*");
+    result = sendCustomSscanCommand(KEY, KEY, ZERO_CURSOR, "MATCH", "3*", "MATCH", "1*");
 
-    assertThat((byte[]) result.get(0)).isEqualTo("0".getBytes());
-    assertThat((List<byte[]>) result.get(1)).containsExactlyInAnyOrder("1".getBytes(),
-        "12".getBytes());
+    assertThat(result.getCursor()).isEqualTo(ZERO_CURSOR);
+    assertThat(result.getResult()).containsOnly(FIELD_ONE.getBytes(),
+        FIELD_TWO.getBytes());
   }
 
   @Test
-  public void givenMatchAndCount_returnsAllMembersWithoutDuplicates() {
-    jedis.sadd("a", "1", "12", "3");
-
+  public void givenSetWithThreeMembersAndMatchAndCount_returnsAllMatchingMembers() {
+    jedis.sadd(KEY, FIELD_ONE, FIELD_TWO, FIELD_THREE);
     ScanParams scanParams = new ScanParams();
     scanParams.count(1);
     scanParams.match("1*");
-    ScanResult<byte[]> result;
     List<byte[]> allMembersFromScan = new ArrayList<>();
-    String cursor = "0";
+    String cursor = ZERO_CURSOR;
 
     do {
-      result = jedis.sscan("a".getBytes(), cursor.getBytes(), scanParams);
+      result = jedis.sscan(KEY.getBytes(), cursor.getBytes(), scanParams);
       allMembersFromScan.addAll(result.getResult());
       cursor = result.getCursor();
     } while (!result.isCompleteIteration());
 
-    assertThat(allMembersFromScan).containsExactlyInAnyOrder("1".getBytes(),
-        "12".getBytes());
+    assertThat(allMembersFromScan).containsOnly(FIELD_ONE.getBytes(),
+        FIELD_TWO.getBytes());
   }
 
   @Test
-  @SuppressWarnings("unchecked")
-  public void givenMultipleCountsAndMatches_returnsAllEntriesWithoutDuplicates() {
-    jedis.sadd("a", "1", "12", "3");
-
-    List<Object> result;
+  public void givenSetWithThreeMembersAndMultipleMatchAndCountArguments_returnsAllMatchingMembers() {
```

Comment:
This test should be modified in a similar way to the one above, but with different COUNT values:
```
  @Test
  public void givenMultipleCountAndMatch_usesLastSpecified() {
    Set<byte[]> initialMemberData = initializeThousandMemberByteSet();

    // Choose a large COUNT to ensure that some matching members are returned
    // There are 111 matching members in the set 0..999
    result = sendCustomSscanCommand(KEY, KEY, ZERO_CURSOR,
        "COUNT", "20",
        "MATCH", "1*",
        "COUNT", "950",
        "MATCH", "9*");

    List<byte[]> returnedMembers = this.result.getResult();
    // We know that we must have found at least 61 matching members, given the size of COUNT and the
    // number of matching members in the set
    assertThat(returnedMembers.size()).isGreaterThanOrEqualTo(61);
    assertThat(returnedMembers).isSubsetOf(initialMemberData);
    assertThat(returnedMembers).allSatisfy(bytes -> assertThat(new String(bytes)).startsWith("9"));
  }
```